File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/permissions/storage/request/StoragePermissionsRequestNotificationBuilder.java
package com.lasthopesoftware.bluewater.client.library.permissions.storage.request;

import android.app.Notification;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.app.NotificationCompat;

import com.lasthopesoftware.bluewater.R;
import com.lasthopesoftware.bluewater.client.settings.EditClientSettingsPendingIntentBuilder;
import com.lasthopesoftware.bluewater.client.settings.IEditClientSettingsPendingIntentBuilder;
import com.lasthopesoftware.resources.strings.IStringResourceProvider;
import com.lasthopesoftware.resources.strings.StringResourceProvider;

/**
 * Created by david on 7/10/16.
 */
public class StoragePermissionsRequestNotificationBuilder implements IStoragePermissionsRequestNotificationBuilder {

	private final NotificationCompat.Builder notificationBuilder;
	private final IStringResourceProvider stringResourceProvider;
	private final IEditClientSettingsPendingIntentBuilder editServerSettingsPendingIntentBuilder;

	public StoragePermissionsRequestNotificationBuilder(Context context) {
		this(new NotificationCompat.Builder(context), new StringResourceProvider(context), new EditClientSettingsPendingIntentBuilder(context));
	}

	public StoragePermissionsRequestNotificationBuilder(@NonNull NotificationCompat.Builder notificationBuilder, @NonNull IStringResourceProvider stringResourceProvider, @NonNull IEditClientSettingsPendingIntentBuilder editServerSettingsPendingIntentBuilder) {
		this.notificationBuilder = notificationBuilder;
		this.stringResourceProvider = stringResourceProvider;
		this.editServerSettingsPendingIntentBuilder = editServerSettingsPendingIntentBuilder;
	}

	@NonNull
	@Override
	public Notification buildStoragePermissionsRequestNotification(int libraryId) {
		notificationBuilder.setSmallIcon(R.drawable.clearstream_logo_dark);
		notificationBuilder.setContentTitle(stringResourceProvider.getString(R.string.permissions_needed));
		notificationBuilder.setContentText(stringResourceProvider.getString(R.string.permissions_needed_launch_settings));

		notificationBuilder.setContentIntent(editServerSettingsPendingIntentBuilder.buildEditServerSettingsPendingIntent(libraryId));

		notificationBuilder.setAutoCancel(true);

		notificationBuilder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC);

		return notificationBuilder.build();
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/permissions/StoragePermissionsRequestNotificationBuilder.kt
package com.lasthopesoftware.bluewater.client.stored.library.permissions

import android.app.Notification
import android.content.Context
import androidx.core.app.NotificationCompat
import com.lasthopesoftware.bluewater.R
import com.lasthopesoftware.bluewater.android.intents.BuildIntents
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.shared.android.notifications.notificationchannel.ChannelConfiguration
import com.lasthopesoftware.resources.strings.GetStringResources

class StoragePermissionsRequestNotificationBuilder(
    private val context: Context,
    private val stringResources: GetStringResources,
    private val intentBuilder: BuildIntents,
    private val channelProperties: ChannelConfiguration,
) : IStoragePermissionsRequestNotificationBuilder {

	override fun buildStoragePermissionsRequestNotification(libraryId: Int): Notification {
		val notificationBuilder = NotificationCompat.Builder(context, channelProperties.channelId)

		notificationBuilder.setSmallIcon(R.drawable.now_playing_status_icon_white)
		notificationBuilder.setContentTitle(stringResources.permissionsNeeded)
		notificationBuilder.setContentText(stringResources.permissionsNeededLaunchSettings)
		notificationBuilder.setContentIntent(
			intentBuilder.buildLibraryServerSettingsPendingIntent(LibraryId(libraryId))
		)
		notificationBuilder.setAutoCancel(true)
		notificationBuilder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
		return notificationBuilder.build()
	}
}
